### PR TITLE
SOC-5514 Space activities are lost after migration when the JCR space identity node has a name not equal to remoteId

### PR DESCRIPTION
**Pull request by nttuyen**
- Make the node name and the remoteId are the same before migrate identity

### CODE REVIEW

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/updater/IdentityMigrationService.java [hunk 1]
```diff
@@ -305,15 +315,51 @@ public void doRemove() throws Exception {
...
     String remoteId = node.getProperty("soc:remoteId").getString();
+    if (!name.equals(remoteId)) {
+      boolean needUpdateRemoteId = true;
+      if (SpaceIdentityProvider.NAME.equals(providerId)) {
+        Space space = spaceStorage.getSpaceByPrettyName(name);
+        if (space == null) {
+          space = spaceStorage.getSpaceByPrettyName(remoteId);
+          // If we can not find the space for this identity, we could ignore to migrate
+          if (space == null) {
+            return null;
```
Comment (boubaker) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/updater/IdentityMigrationService.java]: Add log entry saying that the space was not found and will not be migrated.
This is a very important log entry that should figure in final report.

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/updater/IdentityMigrationService.java [hunk 2]
```diff
@@ -305,15 +315,51 @@ public void doRemove() throws Exception {
 
   private Identity migrateIdentity(Node node, String jcrId) throws Exception {
     String providerId = node.getProperty("soc:providerId").getString();
+    // The node name is the identity id.
+    // Node name is soc:<name>, only the <name> is relevant
```
Comment (boubaker) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/updater/IdentityMigrationService.java]: Add comment to describe what is done when (name != remoteId) and add log.info to trace that we will update this mess.

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/updater/IdentityMigrationService.java [hunk 3]
```diff
@@ -305,15 +315,51 @@ public void doRemove() throws Exception {
...
+          // If we can not find the space for this identity, we could ignore to migrate
+          if (space == null) {
+            return null;
+          } else {
+            needUpdateRemoteId = false;
+            name = remoteId;
+          }
+        }
+      }
+
```
Comment (boubaker) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/updater/IdentityMigrationService.java]: Please add a comment describing what is done here.
If remoteId is not updated => update the property
If name is not coherent => move node

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/updater/IdentityMigrationService.java (hunk 2 above)
Comment (nttuyen) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/updater/IdentityMigrationService.java]: Done

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/updater/IdentityMigrationService.java (hunk 1 above)
Comment (nttuyen) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/updater/IdentityMigrationService.java]: Done

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/updater/IdentityMigrationService.java (hunk 3 above)
Comment (nttuyen) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/updater/IdentityMigrationService.java]: Done

**Pull request merged**